src/main/java/org/cactoos/func/TimedFunc.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.func;
+
+import java.util.concurrent.TimeUnit;
+import org.cactoos.Func;
+import org.cactoos.Proc;
+
+/**
+ * Function that gets interrupted after a certain time has passed.
+ *
+ * @author Vedran Vatavuk (123vgv@gmail.com)
+ * @version $Id$
+ * @param <X> Type of input
+ * @param <Y> Type of output
+ * @since 0.29.3
+ */
+public final class TimedFunc<X, Y> implements Func<X, Y> {
+
+    /**
+     * Origin function.
+     */
+    private final Func<X, Y> origin;
+
+    /**
+     * Milliseconds.
+     */
+    private final long time;
+
+    /**
+     * Ctor.
+     * @param proc Proc
+     * @param milliseconds Milliseconds
+     */
+    public TimedFunc(final Proc<X> proc, final long milliseconds) {
+        this(new FuncOf<>(proc), milliseconds);
+    }
+
+    /**
+     * Ctor.
+     * @param function Origin function
+     * @param milliseconds Milliseconds
+     */
+    public TimedFunc(final Func<X, Y> function, final long milliseconds) {
+        this.origin = function;
+        this.time = milliseconds;
+    }
+
+    @Override
+    public Y apply(final X input) throws Exception {
+        return new AsyncFunc<>(this.origin).apply(input)
```

Comment:
@Vatavuk In the issue description, I see the phrase "interrupted after a certain time has passed". Also in [connected issue](https://github.com/yegor256/cactoos-http/issues/10) there is the phrase "terminate the connection if it's taking too long". So looks like you need to interrupt thread which processes future after a timeout. `AsyncFunc`, which you use, return `Future` and you can call `Future.cancel(true)` for doing it.

Reply:
You could try/catch on `TimeOutException` and cancel it when it happens instead of letting the Exception get thrown back for example.

Reply:
@SharplEr fixed